components/PortalStatistics.lua

```diff
@@ -0,0 +1,1225 @@
+		conditions = addConditions or '',
+		order = addOrder,
+		groupby = addGroupBy,
+		limit = limit or 5000,
+	})
+
+	return data
+end
+
+
+function StatisticsPortal._getTeams(limit, addConditions, addOrder, addGroupBy)
+	local data = mw.ext.LiquipediaDB.lpdb('team', {
+		query = 'pagename, name, template, earnings, extradata',
+		conditions = addConditions or '',
+		order = addOrder,
+		groupby = addGroupBy,
+		limit = limit or 5000,
+	})
+
+	return data
+end
+
+
+function StatisticsPortal._getPieChartData(args, groupBy, defaultValue, groupValues)
+	args = args or {}
+	table.insert(groupValues, defaultValue)
+	defaultValue = string.lower(defaultValue or '')
+
+	local prizes = {}
+	for _, value in Table.iter.spairs(groupValues) do
+		prizes[value:lower()] = {name = value, value = 0}
+	end
+
+	local LPDBConditions = StatisticsPortal._returnBaseConditions()
+	LPDBConditions:add{ConditionNode(ColumnName('namespace'), Comparator.eq, 0)}
+
+	if args.year then
+		LPDBConditions:add{ConditionNode(ColumnName('sortdate_year'), Comparator.eq, args.year)}
+	else
+		LPDBConditions:add{ConditionNode(ColumnName('sortdate'), Comparator.lt, DATE)}
+	end
+
+	if args.game then
+		LPDBConditions:add{ConditionNode(ColumnName('game'), Comparator.eq, args.game)}
+	end
+
+	local function parseTournament(data)
+		local normValue = string.lower(data[groupBy] or '')
+		if prizes[normValue] then
+			prizes[normValue].value = prizes[normValue].value + data.prizepool
+		else
+			prizes[defaultValue].value = prizes[defaultValue].value + data.prizepool
+		end
+	end
+
+	--Querying data
+	local queryParameters = {
+		conditions = LPDBConditions:toString(),
+		query = 'prizepool, '.. groupBy,
+	}
+
+	--Querying data
+	Lpdb.executeMassQuery('tournament', queryParameters, parseTournament)
+
+	Array.forEach(Array.extractValues(prizes), function(prize)
+		prize.value = math.floor(prize.value + 0.5)
+	end)
+
+	if prizes[defaultValue].value == 0 then
+		Table.extract(prizes, defaultValue)
+	end
+
+	local chartData = Array.map(Array.extractValues(groupValues), function(value)
+		return prizes[value:lower()]
+	end)
+
+	return StatisticsPortal._drawPieChart(args, chartData)
+end
+
+
+function StatisticsPortal._cacheModeEarningsData(config)
+	local conditions = ConditionTree(BooleanOperator.all)
+		:add{ConditionNode(ColumnName('prizemoney'), Comparator.gt, 0)}
+		:add{ConditionNode(ColumnName('date'), Comparator.neq, EPOCH_DATE)}
+		:add{ConditionNode(ColumnName('date'), Comparator.lt, DATE)}
+
+	if String.isNotEmpty(config.startYear) then
+		conditions:add{ConditionNode(ColumnName('date_year'), Comparator.gt, (config.startYear - 1))}
+	end
+
+	if String.isNotEmpty(config.opponentName) then
+		local teamConditions = ConditionTree(BooleanOperator.any)
+			:add{ConditionNode(ColumnName('opponentname'), Comparator.eq, config.opponentName)}
+		local prefix = config.opponentType == Opponent.team and 'team' or ''
+		for index = 1, 30 do
```

Comment:
why the 30?
make const var with a name that makes it clear^^

Reply:
I did a quick check of the modules that people were using that were intended to be replaced by this one, and 30 was the highest limit used for opponent players on a team stored in placement, so I used that when initially testing.

The hard limit would be 10 though, if I'm not mistaken? As that's the highest amount that can be stored in the teamcard?

Reply:
the limit is wiki specific most wikis only store up to 10
but some overwrite that
my point was more to make it a const var or possibly even an input value and give it a proper name

Reply:
Should the default be 10 then? If I make it an input parameter to be specified with the module call, with a fallback default value?

Reply:
probably